src/java.base/share/classes/jdk/internal/module/ModuleReferenceImpl.java

```diff
@@ -62,8 +62,22 @@ public class ModuleReferenceImpl extends ModuleReference {
     // ModuleResolution flags
     private final ModuleResolution moduleResolution;
 
-    // cached hash of this module to avoid needing to compute it many times
-    private byte[] cachedHash;
+    // Single-slot cache of this module's hash to avoid needing to compute
+    // it many times. If there is a significant number of hash algorithms
+    // used in the future, consider extending this cache to full
+    // ConcurrentMap<String,byte[]>.
+    //
+    // For correctness under concurrent updates, we need to wrap the fields
```

Comment:
The comment "If there is a significant ..." reads more like a "TODO" and can be removed.

Reply:
Done.